README.rst

```diff
@@ -152,6 +152,110 @@ URL to use to validate and get the token information.
 Connexion expects to receive the Oauth token in the ``Authorization`` header field in the format described in
 `RFC 6750 <rfc6750_>`_ section 2.1.
 
+
+
+Use your own security decorator
+-------------------------------
+Connexion adds a security decorator for each view function to handle (oauth) security. If you need your own security logic
+like checking the token at your own endpoint and or setting a current user in your request context, you can create your own
+security decorator and pass it to connexion at api creation time, like this
+
+.. code-block:: python
+
+    import my_module.security.decorators
+
+    app.add_api('my_api.yaml',
+                security_decorator = my_module.security.decorators.verify_oauth_at_my_way
+                base_path='/1.0')
+
+
+
+Look for example security decorators in `connexion.decorators.security`
+A custom decorator may look like this:
+
+.. code-block:: python
+
+  """
+  override the connexion security decorators
+  """
+
+  # Authentication and authorization related decorators
+
+  from flask import request
+  import functools
+  import logging
+  import requests
+  from connexion import problem
+
+  logger = logging.getLogger('connexion.api.custom_security')
+
+  # use connection pool for OAuth tokeninfo
+  adapter = requests.adapters.HTTPAdapter(pool_connections=100, pool_maxsize=100)
+  session = requests.Session()
+  session.mount('http://', adapter)
+  session.mount('https://', adapter)
+
+
+
+  def verify_oauth_at_my_way(allowed_scopes, function):
+      """
+      Special Decorator to verify oauth and set user and role(s)
+
+      :param token_info_url: Url to get information about the token
+      :type token_info_url: str
+      :param allowed_scopes: Set with scopes that are allowed to access the endpoint
+      :type allowed_scopes: set
+      :type function: types.FunctionType
+      :rtype: types.FunctionType
+      """
+
+      @functools.wraps(function)
+      def wrapper(*args, **kwargs):
```

Comment:
Maybe you could use a way simpler example code than that? It would be better to have as example the very minimal work necessary to have a custom security decorator working. So it could be used as bootstrap code for the reader's code.